internal/models/api/token.go

```diff
@@ -0,0 +1,52 @@
+// Package internal/models/api defines models to be used for API requests.
+package api
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/SimplQ/simplQ-golang/internal/models/db"
+)
+
+// AddTokenRequest is a model to structure an add token request.
+//
+// Members
+//
+// QueueId          -   QueueId of the queue that this token is to be added to
+// Name             -   Name of the token, typically name of the person whom the
+//                      token was issued to.
+// ContactNumber    -   Contact Number.
+// EmailId          -   Optional. Email ID if the queue collects email ID of users.
+type AddTokenRequest struct {
+	QueueId       db.QueueId
+	Name          string
+	ContactNumber string
+	EmailId       string
+}
+
+// AddTokenResponse is a model to structure the response of an add token request.
+type AddTokenResponse db.TokenId
+
+func (req AddTokenRequest) Validate() (ValidationError, bool) {
+	if len(req.Name) < MIN_LENGTH || len(req.Name) > MAX_LENGTH {
+		message := fmt.Sprintf("Token name length should be greater than %d characters and less than %d characters", MIN_LENGTH, MAX_LENGTH)
+		return ValidationError{
+			Field:   "Name",
+			Message: message,
+		}, false
+	} else if len(req.ContactNumber) != 10 {
+		message := fmt.Sprintf("Contact number should be 10 digits")
+		return ValidationError{
+			Field:   "ContactNumber",
+			Message: message,
+		}, false
+	} else if _, err := strconv.Atoi(req.ContactNumber); err != nil {
+		message := fmt.Sprintf("Contact number should only consist of digits")
+		return ValidationError{
+			Field:   "ContactNumber",
+			Message: message,
+		}, false
```

Comment:
This is how a phone number from frontend could currently look like:
<img width="1379" alt="image" src="https://user-images.githubusercontent.com/11256376/154828096-49f2a5d4-185d-409d-98bc-1035dad25ada.png">

Reply:
Let's use this same library that the frontend uses: https://github.com/ttacon/libphonenumber

Reply:
The default `Parse` function seems to just parse any string ignoring invalid characters. Only giving an error when there are no numbers. Even translates alphabets to numbers. Is it supposed to be that way? If the frontend already verifies phone numbers this shouldn't be an issue.

Also, Do we also use this to parse the given string and store that in the DB? Or just to verify?

Reply:
The frontend does verify phone numbers. What do you mean by translates alphabets to numbers? 

I believe we do store the number in the DB.